In collection, give changed status in workflow_job_template when destroying nodes

sean-m-sullivan opened the pull request:
##### SUMMARY
makes destroy current nodes schema report changed status correctly
Resolves #13387

##### ISSUE TYPE
 - New or Enhanced Feature

##### COMPONENT NAME
 - Collection


##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
22.1.0
```

sean-m-sullivan committed:
make collection workflow job template report correct changed status on workflow node destruction.

AlanCoding commented:
This doesn't look coherent for the case where it is _not_ an SCM inventory import. Those inventory updates don't use `scm_revision` at all, and thus, it will probably be blank. That suggests to me that this would have ec2 updates, for example, always give changed=False which isn't desirable.

Even for SCM inventory updates, the revision is only valid as a metric for changed status if the content it static, which is often isn't. Cloud sources can change.

sean-m-sullivan commented:
@AlanCoding I did not take that into account you are correct, I've renamed the PR, and removed the inventory source update code, and just left the workflow node change.
